### PR TITLE
fix(web): Fix status of member's setting page

### PR DESCRIPTION
**Pull request by ZTongci**
# Overview

## Summary by CodeRabbit

- **New Features**
	- Introduced a new modal for adding members to a workspace with search functionality.
	- Added a modal for updating team member roles within a workspace.
	- Implemented a confirmation modal for deleting workspaces, enhancing user feedback.
	- Added a new component for managing workspace settings, including updating names and handling deletions.
	- Redesigned the Members component to streamline member management within a workspace.

- **Bug Fixes**
	- Improved error handling and user notifications for workspace management actions.

- **Refactor**
	- Streamlined the workspace settings component for better performance and usability.
	- Simplified the logic for rendering workspace-related components, enhancing clarity.
	- Updated translation files to improve clarity and user guidance for workspace functionalities.

### CODE REVIEW

File: web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx [hunk 1]
```diff
@@ -158,62 +135,85 @@ const Members: FC<Props> = ({
...
+            )
+          } as Workspace;
+        });
+      }
       setActiveEditIndex((prevIndex) => (prevIndex === index ? null : index));
     }
   };
 
-  const handleRemoveMemberButtonClick = (userId: string) => {
+  const handleRemoveMemberButtonClick = async (userId: string) => {
```
Comment (mkumbobeaty) [inline web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx:167]: you can wrap this with useCallback

File: web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx [hunk 2]
```diff
@@ -158,62 +135,85 @@ const Members: FC<Props> = ({
...
+            members: prevMembers?.members?.filter(
+              (workspaceMember) => workspaceMember.userId !== userId
+            )
+          } as Workspace;
+        });
+      }
     }
   };
 
   const handleAddMember = () => {
```
Comment (mkumbobeaty) [inline web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx:186]:
```suggestion
  const handleAddMember = useCallback() => {
```

File: web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx [hunk 3]
```diff
@@ -158,62 +135,85 @@ const Members: FC<Props> = ({
...
-        ]);
+
+        if (status === "success") {
+          setCurrentWorkspace((prevMembers) => {
+            return {
+              ...prevMembers,
+              members: [
+                ...(prevMembers?.members || []),
+                {
+                  __typename: "TeamMember",
```
Comment (mkumbobeaty) [inline web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx:203]: Why do we need to define this??

File: web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx [hunk 4]
```diff
@@ -158,62 +135,85 @@ const Members: FC<Props> = ({
...
+                  user: {
+                    email: memberSearchResult.email,
+                    id: memberSearchResult.id,
+                    name: memberSearchResult.userName,
+                    __typename: "User"
```
Comment (mkumbobeaty) [inline web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx:208]: why this  __typename is defined?

File: web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx (hunk 3 above)
Comment (airslice) [reply web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx]: I did some refactor and this part is outdated, i'll close this conversation. Thank you!

File: web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx (hunk 4 above)
Comment (airslice) [reply web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx]: I did some refactor and this part is outdated, i'll close this conversation. Thank you!

**Review by airslice — APPROVED** [review]

**Pull request merged**